utilities/rpp-unittests/OCL/BatchPD_ocl_pkd3.cpp

```diff
@@ -0,0 +1,2130 @@
+                }
+            }
+        }
+        i++;
+    }
+    closedir(dr2);
+
+    if (ip_bitDepth == 1)
+    {
+        Rpp8u *inputTemp, *input_secondTemp;
+        Rpp16f *inputf16Temp, *inputf16_secondTemp;
+       
+        inputTemp = input;
+        input_secondTemp = input_second;
+
+        inputf16Temp = inputf16;
+        inputf16_secondTemp = inputf16_second;
+
+        for (int i = 0; i < ioBufferSize; i++)
+        {
+            *inputf16Temp = (Rpp16f) *inputTemp / 255.0;
+            *inputf16_secondTemp = (Rpp16f) *input_secondTemp / 255.0;
+            inputTemp++;
+            inputf16Temp++;
+            input_secondTemp++;
+            inputf16_secondTemp++;
+        }
+    }
+    else if (ip_bitDepth == 2)
+    {
+        Rpp8u *inputTemp, *input_secondTemp;
+        Rpp32f *inputf32Temp, *inputf32_secondTemp;
+       
+        inputTemp = input;
+        input_secondTemp = input_second;
+
+        inputf32Temp = inputf32;
+        inputf32_secondTemp = inputf32_second;
+
+        for (int i = 0; i < ioBufferSize; i++)
+        {
+            *inputf32Temp = (Rpp32f) *inputTemp / 255.0;
+            *inputf32_secondTemp = (Rpp32f) *input_secondTemp / 255.0;
+            inputTemp++;
+            inputf32Temp++;
+            input_secondTemp++;
+            inputf32_secondTemp++;
+        }
+    }
+
+    cl_mem d_input,d_inputf16,d_inputf32, d_input_second, d_output,d_outputf16,d_outputf32;
+cl_platform_id platform_id;
+cl_device_id device_id;
+cl_context theContext;
+cl_command_queue theQueue;
+cl_int err;
+    err = clGetPlatformIDs(1, &platform_id, NULL);
+    err = clGetDeviceIDs(platform_id, CL_DEVICE_TYPE_GPU, 1, &device_id, NULL);
+    theContext = clCreateContext(0, 1, &device_id, NULL, NULL, &err);
+    theQueue = clCreateCommandQueue(theContext, device_id, 0, &err);
+    d_input = clCreateBuffer(theContext, CL_MEM_READ_ONLY, ioBufferSize * sizeof(Rpp8u), NULL, NULL);
+    d_inputf16 = clCreateBuffer(theContext, CL_MEM_READ_ONLY, ioBufferSize * sizeof(Rpp16f), NULL, NULL);
+    d_inputf32 = clCreateBuffer(theContext, CL_MEM_READ_ONLY, ioBufferSize * sizeof(Rpp32f), NULL, NULL);
+    d_input_second = clCreateBuffer(theContext, CL_MEM_READ_ONLY, ioBufferSize * sizeof(Rpp8u), NULL, NULL);
+    d_output = clCreateBuffer(theContext, CL_MEM_READ_ONLY, oBufferSize * sizeof(Rpp8u), NULL, NULL);
+    d_outputf16 = clCreateBuffer(theContext, CL_MEM_READ_ONLY, oBufferSize * sizeof(Rpp16f), NULL, NULL);
+    d_outputf32 = clCreateBuffer(theContext, CL_MEM_READ_ONLY, oBufferSize * sizeof(Rpp32f), NULL, NULL);
+    err = clEnqueueWriteBuffer(theQueue, d_input, CL_TRUE, 0, ioBufferSize * sizeof(Rpp8u), input, 0, NULL, NULL);
+    err = clEnqueueWriteBuffer(theQueue, d_inputf16, CL_TRUE, 0, ioBufferSize * sizeof(Rpp16f), inputf16, 0, NULL, NULL);
+    err = clEnqueueWriteBuffer(theQueue, d_inputf32, CL_TRUE, 0, ioBufferSize * sizeof(Rpp32f), inputf32, 0, NULL, NULL);
```

Comment:
Codacy found an issue: [Variable 'err' is reassigned a value before the old one has been used.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/rpp/pullRequest?prid=5976402)